pennylane/templates/layers/__init__.py

```diff
@@ -21,3 +21,4 @@
 from .cv_neural_net import CVNeuralNetLayers
 from .simplified_two_design import SimplifiedTwoDesign
 from .basic_entangler import BasicEntanglerLayers
+from .arbitrary_unitary import ArbitraryUnitary
```

Comment:
I'm not sure I understand why this is a layer? I would consider this a subroutine, because people are unlikely repeating it the template like other layers, and it is currently not really implemented (since ``angles`` does not use the first dimension to indicate the number of layers).

Reply:
I kinda feel like making too fine-grained classifications is not really worth it, especially when these concepts haven't fully crystallized in the community. I'd be in favour of everything (embeddings, layers, subroutines, ...) just living under `qml.templates` for simplicity

Reply:
I don't know whats best and leave this decision to you. I did use it as a lay in few-qubit (3) simulations, but beyond that the number of parameters really explodes, so this might be better as a building block, e.g. simulating an arbitrary 2-qubit gate.

Reply:
Yes @johannesjmeyer . 

Can we for now put it into "subroutines" please? Layers all have weight arguments where the first dimension is the number of layers... 

The distinctions are quite crucial for the upcoming templates class, they are not totally random :)